File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/transactions/ui/detail/ui/TransactionDetailViewComponent.java
package com.fintechplatform.ui.transactions.ui.detail.ui;

import com.fintechplatform.ui.email.SendEmailHelperModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component (modules = {
        TransactionDetailPresenterModule.class,
        TransactionDetailUIModule.class,
        SendEmailHelperModule.class
})
public interface TransactionDetailViewComponent {
    void inject(TransactionDetailActivity activity);
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/transactions/ui/detail/di/TransactionDetailViewComponent.kt
package com.fintechplatform.ui.transactions.ui.detail.di

import com.fintechplatform.ui.email.SendEmailHelperModule
import com.fintechplatform.ui.transactions.ui.detail.TransactionDetailFragment
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [TransactionDetailPresenterModule::class, SendEmailHelperModule::class])
interface TransactionDetailViewComponent {
    fun inject(fragment: TransactionDetailFragment)
}
